### DISCUSSION TITLE
Templater Error: Invalid reference date format, try specifying one with the argument ‘reference_format’.

### DISCUSSION THREAD

**Discussion by Angelikini** [SilentVoid13/Templater#755] [Help]
Hello,
I am trying to build a template for a daily note and I get the following error:

Templater Error: Invalid reference date format, try specifying one with the argument ‘reference_format’.

Here is the template:

### To do:
```tasks
not done
due <%+ tp.date.now("YYYY-MM-DD", tp.file.title, "YYYY-MM-DD") %>
```

### Done
```tasks
done on <%+ tp.date.now("YYYY-MM-DD",tp.file.title, "YYYY-MM-DD")%>
```
The file name is in the format YYY-MM-DD

Any idea what I am doing wrong?

**Comment by shabegom** [comment]
Can you provide an example of the file title? This should, in theory, work.

**Comment by Seraphli** [comment]
Hello, I'm facing the same problem here. I use this template and set the corresponding location.
```
--- 

Creation date: <% tp.file.creation_date() %> 
Modification date: <% tp.file.last_modified_date("YYYY-MM-DD HH:mm:ss") %> 

---

<< [[<% tp.date.now("YYYY-MM-DD", -1) %>]] | [[<% tp.date.now("YYYY-MM-DD", 1) %>]] >>
# <% tp.date.now("YYYY-MM-DD", 0, tp.file.title, "YYYY-MM-DD") %>
<% tp.file.cursor() %>
```
When I create a daily note with the `Daily Note` plugin or `Calender` plugin, this template is working.
But when I create using `right click -> new note` in the folder, the error pops out.

**Comment by Seraphli** [comment]
I manage to debug the plugin and find out the title using `right click -> new note` is `Untitled`, and it can not be converted to a datetime. And the daily note plugin passes the workable title to templater, so that way works.

**Comment by Seraphli** [comment]
Just a note for someone, the working template is this
```
--- 

Creation date: <% tp.file.creation_date() %> 
Modification date: <%+ tp.file.last_modified_date("YYYY-MM-DD HH:mm:ss") %> 

---

<< [[<% tp.date.now("YYYY-MM-DD", -1) %>]] | [[<% tp.date.now("YYYY-MM-DD", 1) %>]] >>
<%- tp.file.rename(tp.date.now()) %>
# <% tp.date.now("YYYY-MM-DD", 0) %>
<% tp.file.cursor() %>
```
It will auto rename your note with a datetime string. But this can not deal with the situation when the file already exists. I don't know how to cancel the creation when the destination file already exists.
